tests/integration/run_integration_tests.sh

```diff
@@ -0,0 +1,222 @@
+      fi
+    ;;
+    r)
+      if [[ "${INTEGRATION_TESTS_TO_SKIP:-"not-set"}" == "not-set" ]]; then
+      readarray -d ' ' -t INTEGRATION_TESTS_TO_RUN <<< "${OPTARG}"
+      else
+        echo "Aborting: Please specify either the run-only (-r) or the skip (-s) flag, not both."
+        exit 1
+      fi
+    ;;
+    c)
+      CLUSTER="${OPTARG}"
+      if ! [[ "${CLUSTER}" =~ ^(minikube|kind)$ ]]; then
+        echo "Aborting: This script only supports local integration tests with either minikube or kind."
+        exit 1
+      fi
+    ;;
+    p)
+      DESTINATION="${OPTARG}"
+    ;;
+    e)
+      EXTENDED="true"
+  esac
+done
+shift
+
+mkdir tmp
+
+export COSIGN_PUBLIC_KEY="-----BEGIN PUBLIC KEY-----<br>MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEQ70fJ/NA109p3/43cTgwnLQ+HsvK<br>jFK/0L1kLUQO+e2cL+3+y2s4IykwTxfSnqOSA0kEy1rTrhrXLCox4uRmSQ==<br>-----END PUBLIC KEY-----"
+
+if [[ "${CLUSTER:-"minikube"}" == "minikube" ]]; then
+  if [[ "$(docker inspect minikube > /dev/null 2>&1 | jq -r .[].State.Status || echo 'container not found')" != "running" ]]; then
+    echo "Minikube cluster configured. You need a running minikube cluster. Exiting..."
+    exit 1
+  fi
+  eval $(minikube docker-env)
+  echo "Building docker image..."
+  make docker > /dev/null
+  echo "Done building docker image!"
+  NETWORK=$(docker container inspect minikube | jq -r '.[].NetworkSettings.Networks | to_entries | .[].key')
+else
+  if [[ "$(docker inspect kind-control-plane > /dev/null 2>&1 | jq -r .[].State.Status || echo 'container not found')" != "running" ]]; then
+    echo "Kind cluster configured. You need a running kind cluster. Exiting..."
+    exit 1
+  fi
+  echo "Building docker image..."
+  make docker > /dev/null
+  echo "Done building docker image!"
+  kind load docker-image $(yq e '.kubernetes.deployment.image.repository' helm/values.yaml):v$(yq e '.appVersion' helm/Chart.yaml)
+  NETWORK=$(docker container inspect kind-control-plane | jq -r '.[].NetworkSettings.Networks | to_entries | .[].key')
+fi
+
+echo "Spinning up mocked alerting interface..."
+docker run -d --name alerting-endpoint -p 56243:56243 docker.io/securesystemsengineering/alerting-endpoint:latest
+docker network connect ${NETWORK} alerting-endpoint
+export ALERTING_ENDPOINT_IP=$(docker container inspect alerting-endpoint | jq -r --arg network ${NETWORK} '.[].NetworkSettings.Networks[$network].IPAddress')
+echo "Done spinning up alerting interface!"
+
+readarray -t FULL_INTEGRATION_TEST_SUITE < <(yq e -r -o=j '.jobs.[].strategy.matrix | select(.integration-test-arg) | .integration-test-arg[]' .github/workflows/.reusable-integration-test.yml)
+
+FAILED_INTEGRATION_TESTS=()
+SUCCESSFUL_INTEGRATION_TESTS=()
+SCRIPT_FAILURES=()
+
+INTEGRATION_TEST_SUITE=( "${INTEGRATION_TESTS_TO_RUN[@]:-${FULL_INTEGRATION_TEST_SUITE[@]}}" )
+
+if [[ ${EXTENDED} == "true" || "${INTEGRATION_TESTS_TO_RUN[@]}" =~ "self-hosted-notary" ]]; then
```

Comment:
Not that it really makes a difference, but semantically, wouldn't it be
```suggestion
if [[ ${EXTENDED} == "true" || "${INTEGRATION_TEST_SUITE[@]}" =~ "self-hosted-notary" ]]; then
```
?

Reply:
Nope, because that should evaluate to true only if the user specifically provided `self-hosted-notary` via `-r`  flag.

Reply:
Do you dislike that default?

Reply:
I don't really care. To me it just looked like `self-hosted-notary` wouldn't be part of the `FULL_INTEGRATION_TEST_SUITE` default anyway, so this way even if we decided to add it to defaults in the CI, this script would be more opinionated on it definitely not being default, which I'm totally fine with, just wasn't sure whether it was intentional